pkg/controller/notification/notification_controller.go

```diff
@@ -51,8 +55,15 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 }
 
 // newReconciler returns a new reconcile.Reconciler
-func newReconciler(mgr manager.Manager, config *configuration.Config) reconcile.Reconciler {
-	return &ReconcileNotification{client: mgr.GetClient(), scheme: mgr.GetScheme(), config: config}
+func newReconciler(mgr manager.Manager, config *configuration.Config) (reconcile.Reconciler, error) {
+	factory := NewNotificationDeliveryServiceFactory(mgr.GetClient(), config, config)
```

Comment:
it seems a bit weird that we have to pass the `config` twice in this call. Would it be possible to merge the `NotificationDeliveryServiceConfig` and `MailgunConfiguration` interfaces for example (or have a new interface composed of those 2 one)?

Reply:
good idea! changed as part of: 057eb87e2123a2edfa90a49c007140b845d59010

Reply:
done here: https://github.com/codeready-toolchain/host-operator/pull/251